contracts/exchange/contracts/src/mixins/MExchangeRichErrors.sol

```diff
@@ -0,0 +1,210 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+
+import "@0x/contracts-utils/contracts/src/mixins/MRichErrors.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+
+
+contract MExchangeRichErrors is
+    MRichErrors
+{
+    enum FillErrorCodes {
```

Comment:
It could be beneficial to use a single enum so that errors have a unique identifier across the whole system.

Reply:
If this id were in at the same offset in each message (like the first param) then it would also be easy to identify the error without decoding it, if that would be useful.

Reply:
Mm although I suppose the combination of selector + error code is unique ya?

Reply:
So I think it's probably not necessary to have globally unique error codes because, as you said, the selector is already pretty unique. But, where error codes *are* used, the code ought to be the first parameter. This might be the more language agnostic play since a lazy client can just read the first 36 bytes (selector + code) to get a basic understanding of the error.